build-support/migration-support/modernize_sources.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python3
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+"""A script to modernize the `sources` and `bundles` fields of BUILD files to use the new...
+instead of the deprecated `globs`, `rglobs`, and `zglobs` style.
+
+Run `python3 modernize_sources.py --help`.
+"""
+
```

Comment:
I should point out https://github.com/pantsbuild/pants/blob/master/contrib/buildrefactor/src/python/pants/contrib/buildrefactor/buildozer.py probably...

Modern versions of `buildozer` allow for roughly:
```
$ buildozer 'print sources' src/scala/com/example:example
rglobs("*.scala")
$ buildozer 'remove sources' src/scala/com/example:example
fixed src/scala/com/example/BUILD
$ buildozer 'add sources "*.scala"' src/scala/com/example:example
fixed src/scala/com/example/BUILD
$ diff
-    sources = rglobs("*.scala"),
+    sources = ["*.scala"],
```

see https://github.com/bazelbuild/buildtools/blob/master/buildozer/README.md. A caveat is that it will reformat files files for you, and doesn't support refactoring syntax that is illegal in bazel (imports, for example).

Reply:
> $ buildozer 'print sources' src/scala/com/example:example
rglobs("*.scala")
$ buildozer 'remove sources' src/scala/com/example:example
fixed src/scala/com/example/BUILD
$ buildozer 'add sources "*.scala"' src/scala/com/example:example
fixed src/scala/com/example/BUILD
$ diff
-    sources = rglobs("*.scala"),
+    sources = ["*.scala"],

This change is invalid! We need to expand this to `['**/*.scala']`.

Reply:
Understood... was just an example. Would still need the rewrite in the middle.